src/vm/dllimportcallback.cpp

```diff
@@ -1408,16 +1421,10 @@ VOID UMThunkMarshInfo::RunTimeInit()
         new (&sigInfo) PInvokeStaticSigInfo(pMD);
     else
         new (&sigInfo) PInvokeStaticSigInfo(GetSignature(), GetModule());
-
-    if (sigInfo.GetCallConv() == pmCallConvCdecl)
-    {
-        // caller pop
-        m_cbRetPop = 0;
-    }
-    else
+    if (sigInfo.GetCallConv() != pmCallConvCdecl)
     {
-        // callee pop
-        m_cbRetPop = static_cast<UINT16>(m_cbActualArgSize);
+        // For all the other calling convention except cdecl, callee pops the stack arguments
+        m_cbRetPop += static_cast<UINT16>(m_cbActualArgSize);
```

Comment:
It would be better to keep track of cbRetPop in a local and assign it just once at the end.